backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,342 @@
+            ITraitsProvider traitsProvider) : base(latestLayerChangeModel)
+        {
+            this.relationModel = relationModel;
+            this.ciModel = ciModel;
+            this.effectiveTraitModel = effectiveTraitModel;
+            this.layerModel = layerModel;
+            this.attributeModel = attributeModel;
+            this.traitsProvider = traitsProvider;
+        }
+
+        const int moduleGroupPrio = 10;
+        const int customersGroupPrio = 100;
+        const int networkSegmentGroupPrio = 1000;
+        const int networkInterfaceGroupPrio = 2000;
+        const int assignmentGroupDefaultPrio = 10000;
+        const int baseCIPrio = 100000;
+
+        public override async Task<bool> Run(Layer targetLayer, JObject config, IChangesetProxy changesetProxy, IModelContext trans, ILogger logger)
+        {
+            logger.LogDebug("Start VariableRendering");
+
+            Configuration cfg;
+
+            try
+            {
+                cfg = config.ToObject<Configuration>();
+                logger.LogDebug("Parsed configuration for VariableRendering.");
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("An error ocurred while creating configuration instance.", ex);
+                return false;
+            }
+
+            var layersetVariableRendering = await layerModel.BuildLayerSet(cfg.InputLayerSet.ToArray(), trans);
+
+            var allCIs = await ciModel.GetMergedCIs(new AllCIIDsSelection(), layersetVariableRendering, false, AllAttributeSelection.Instance, trans, changesetProxy.TimeThreshold);
+
+            var activeTrait = await traitsProvider.GetActiveTrait(cfg.BaseCI.RequiredTrait, trans, changesetProxy.TimeThreshold);
+
+            if (activeTrait == null)
+            {
+                logger.LogError($"Could not find baseCI required trait {cfg.BaseCI.RequiredTrait}");
+                return false;
+            }
+
+            var mainCIs = await effectiveTraitModel.FilterCIsWithTrait(allCIs, activeTrait, layersetVariableRendering, trans, changesetProxy.TimeThreshold);
+
+            //TODO: select only the realtions that are defined in configuration
+            //      check if selection with specific predicates is possible
+            var relations = new List<string>();
+
+            cfg.BaseCI.FollowRelations.ForEach(r =>
+            {
+                r.Follow.ForEach(f =>
+                {
+                    if (relations.IndexOf(f.Predicate[1..]) == -1)
+                    {
+                        relations.Add(f.Predicate[1..]);
+                    }
+                });
+            });
+
+            var allRelations = new List<Relation>();
+
+            foreach (var layer in cfg.InputLayerSet)
+            {
+                var layerRelations = await relationModel.GetRelations(RelationSelectionAll.Instance, layer, trans, changesetProxy.TimeThreshold);
+
+                allRelations.AddRange(layerRelations);
+            }
```

Comment:
What is true for fetching CIs/attributes is also true for relations here: do not loop through the input layerset, get the merged relations from the complete layer set